src/core_tests.rs

```diff
@@ -264,18 +264,17 @@ fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
     loop {
         let mut n = false;
         if BALANCED_POLLING {
-            nodes.iter_mut().foreach(|node| n = n || node.inner.poll());
+            // handle at most one event from each node:
```

Comment:
I think [`node.inner.poll()`](https://github.com/Fraser999/routing/blob/master/src/core_tests.rs#L157-L166) is handling _all_ events from each node isn't it?